Lib/uuid.py

```diff
@@ -719,14 +729,48 @@ def uuid5(namespace, name):
     hash = sha1(namespace.bytes + name).digest()
     return UUID(bytes=hash[:16], version=5)
 
+_last_timestamp_v6 = None
+
+def uuid6(node=None, clock_seq=None):
+    """Similar to :func:`uuid1` but where fields are ordered differently
+    for improved DB locality.
+
+    More precisely, given a 60-bit timestamp value as specified for UUIDv1,
+    for UUIDv6 the first 48 most significant bits are stored first, followed
+    by the 4-bit version (same position), followed by the remaining 12 bits
+    of the original 60-bit timestamp.
+    """
+    global _last_timestamp_v6
```

Comment:
`uuid1` uses system-provided UUID generator when available. Maybe it's a good idea to include it here as well?

I know there is no system API for UUID v6 so far, but using the v1 generator and just reversing the byte order to match the v6 + replacing the version number should work here. The drawback is: when using both v1 and v6 UUIDs, it will not generate them concurrently.

Reply:
If a system provided UUIDv6 should exist, then an API should exist as well. In this PR at least, I don't want to introduce a C implementation.

Also, I prefer not to use the same generator for two different UUIDs since they are quite opaque functions (maybe they are not meant to be used like that, but there is no way that we know about that in advance).